server/chat.ts

```diff
@@ -1647,6 +1652,13 @@ export const Chat = new class {
 			entry.hasRoomPermissions = /\bthis\.can\([^,)\n]*, [^,)\n]*,/.test(handlerCode);
 			entry.broadcastable = /\bthis\.(?:canBroadcast|runBroadcast)\(/.test(handlerCode);
 
+			// assign SOME properties from the base command if the current command uses CommandContext.run.
+			const runsCommand = /this.run\(('|"|`)(.*)('|"|`)\)/.exec(handlerCode);
+			if (runsCommand) {
+				const [, , baseCommand] = runsCommand;
+				if (commandTable[baseCommand]) Object.assign(entry, commandTable[baseCommand]);
```

Comment:
That's not just "some", that's all of them. I'm surprised this doesn't crash, actually; I guess the read-only ones are enumerable. But you should copy them one by one instead of using `Object.assign`.